kedro/extras/datasets/api/api_dataset.py

```diff
@@ -38,58 +44,35 @@ def __init__(
-        data: Any = None,
-        params: Dict[str, Any] = None,
-        headers: Dict[str, Any] = None,
-        auth: Union[Iterable[str], AuthBase] = None,
-        json: Union[List, Dict[str, Any]] = None,
-        timeout: int = 60,
-        credentials: Union[Iterable[str], AuthBase] = None,
+        auth_type: str = "requests.auth.HTTPBasicAuth",
+        load_args: Dict[str, Any] = None,
+        credentials: Dict[str, Any] = None,
     ) -> None:
         """Creates a new instance of ``APIDataSet`` to fetch data from an API endpoint.
 
         Args:
             url: The API URL endpoint.
             method: The Method of the request, GET, POST, PUT, DELETE, HEAD, etc...
-            data: The request payload, used for POST, PUT, etc requests
-                https://requests.readthedocs.io/en/master/user/quickstart/#more-complicated-post-requests
-            params: The url parameters of the API.
-                https://requests.readthedocs.io/en/master/user/quickstart/#passing-parameters-in-urls
-            headers: The HTTP headers.
-                https://requests.readthedocs.io/en/master/user/quickstart/#custom-headers
-            auth: Anything ``requests`` accepts. Normally it's either ``('login', 'password')``,
-                or ``AuthBase``, ``HTTPBasicAuth`` instance for more complex cases. Any
-                iterable will be cast to a tuple.
-            json: The request payload, used for POST, PUT, etc requests, passed in
-                to the json kwarg in the requests object.
-                https://requests.readthedocs.io/en/master/user/quickstart/#more-complicated-post-requests
-            timeout: The wait time in seconds for a response, defaults to 1 minute.
-                https://requests.readthedocs.io/en/master/user/quickstart/#timeouts
-            credentials: same as ``auth``. Allows specifying ``auth`` secrets in
-                credentials.yml.
-
-        Raises:
-            ValueError: if both ``credentials`` and ``auth`` are specified.
+            load_args: Additional parameters to be fed to requests.request.
+                https://docs.python-requests.org/en/latest/api/
+            auth_type: provide type to construct a Requests `BaseAuth` object.
+            credentials: Allows specifying secrets in credentials.yml.
         """
         super().__init__()
 
-        if credentials is not None and auth is not None:
-            raise ValueError("Cannot specify both auth and credentials.")
-
-        auth = credentials or auth
+        self._credentials = deepcopy(_DEFAULT_CREDENTIALS)
+        if credentials is not None:
+            self._credentials.update(credentials)
 
-        if isinstance(auth, Iterable):
-            auth = tuple(auth)
+        self._auth = None
+        if credentials is not None:
+            self._auth = create_authenticator(class_type=auth_type, **credentials)
 
         self._request_args: Dict[str, Any] = {
+            **(load_args or {}),
```

Comment:
Minor: could we put this after the other arguments? Just feels a bit weird to me to put the `**kwargs` first here when at least `url` and `method` are compulsory arguments.